### ISSUE TITLE
include-module-naming-schemes doesn't seem to work

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#1367]
I tried to use the option `--include-module-naming-schemes=toolchain_mns.py` with the naming scheme of #1168 but I get the following:
```
-bash-4.2$ eb --include-module-naming-schemes=~/toolchain_mns.py --module-naming-scheme=ToolchainMNS
Usage: eb [options] easyconfig [...]

eb: error: option --module-naming-scheme: invalid choice: 'ToolchainMNS' (choose from 'CategorizedHMNS', 'EasyBuildMNS', 'HierarchicalMNS', 'MigrateFromEBToHMNS')

```

**Comment by boegel** [comment]
@ocaisa: can you try giving it a full path, e.g. `$PWD/toolchain_mns.py`?

Does that file exist?

**Comment by ocaisa** [comment]
Doesn't help, the new MNS doesn't seem to appear in the list of possible options (am I getting the option syntax wrong?). For example, if I try to use an MNS I know doesn't exist, I should get the list of possibilities and the new MNS is not included:
```
-bash-4.2$ eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --module-naming-scheme=test
Usage: eb [options] easyconfig [...]

eb: error: option --module-naming-scheme: invalid choice: 'test' (choose from 'CategorizedHMNS', 'EasyBuildMNS', 'HierarchicalMNS', 'MigrateFromEBToHMNS')

```

**Comment by boegel** [comment]
Can you try enabling 'early debugging' by defining `$DEBUG_EASYBUILD_OPTIONS=1` and provide the output you get?

**Comment by ocaisa** [comment]
Ok the issue occurs when the environment variable EASYBUILD_MODULE_NAMING_SCHEME is set to point to the included MNS:
```
-bash-4.2$  eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --avail-module-naming-schemesList of supported module naming schemes:
	CategorizedHMNS
	EasyBuildMNS
	MigrateFromEBToHMNS
	HierarchicalMNS
	ToolchainMNS
-bash-4.2$ EASYBUILD_MODULE_NAMING_SCHEME=EasyBuildMNS eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --avail-module-naming-schemes
List of supported module naming schemes:
	CategorizedHMNS
	EasyBuildMNS
	HierarchicalMNS
	MigrateFromEBToHMNS
	ToolchainMNS
-bash-4.2$ EASYBUILD_MODULE_NAMING_SCHEME=ToolchainMNS eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --avail-module-naming-schemes
Usage: eb [options] easyconfig [...]

eb: error: option --module-naming-scheme: invalid choice: 'ToolchainMNS' (choose from 'CategorizedHMNS', 'EasyBuildMNS', 'HierarchicalMNS', 'MigrateFromEBToHMNS')

```

**Comment by ocaisa** [comment]
Actually it's really broken because you can't specify that you want to use the scheme at all!
```
-bash-4.2$  eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --module-naming-scheme=EasyBuildMNS --avail-module-naming-schemes
List of supported module naming schemes:
	HierarchicalMNS
	EasyBuildMNS
	MigrateFromEBToHMNS
	CategorizedHMNS
	ToolchainMNS
-bash-4.2$  eb --include-module-naming-schemes=/homeb/zam/swmanage/MNS_TEST/toolchain_mns.py --module-naming-scheme=ToolchainMNS --avail-module-naming-schemes
Usage: eb [options] easyconfig [...]

eb: error: option --module-naming-scheme: invalid choice: 'ToolchainMNS' (choose from 'CategorizedHMNS', 'EasyBuildMNS', 'HierarchicalMNS', 'MigrateFromEBToHMNS')
-bash-4.2$
```

**Comment by boegel** [comment]
Hmm, weird that the unit tests don't trigger this... I guess it's the option parser being a little bit too strict; it probably doesn't know that you're extending the possible choices yet via `--include-module-naming-schemes`, so it bails before the include actually happens...

**Comment by boegel** [comment]
fixed in #1370, let's follow up there
